src/core/public/http/response.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IHttpResponse } from './types';
+
+export class HttpResponse<TResponseBody = any> implements IHttpResponse<TResponseBody> {
+  public readonly request: Request;
+  public readonly response?: Response;
+  public readonly body?: TResponseBody;
+
+  constructor({
+    request,
+    response,
+    body,
+  }: {
+    request: Request;
+    response?: Response;
+    body?: TResponseBody;
```

Comment:
Should we return `body` as a stream? To make it compatible with Response.body https://developer.mozilla.org/en-US/docs/Web/API/Body/body ?